### DISCUSSION TITLE
How to add a new field which starts with hash symbol (#) in remap transform?

### DISCUSSION THREAD

**Discussion by podshivalovdv** [vectordotdev/vector#14191] [Q&A] [answered]
Hi, 
I'm trying to send logs from Vector to Humio and Humio has special meaning for fields which start with the hash symbol `#`, for example `#environment` etc.

I'm using `remap` transform and it looks like that if I use something like this 
```
  docker_generic_logs_transform:
    type: remap
    inputs:
      - docker_db_logs_label
    source: |-
      ""
      log("start")
      .#type= "docker_generic"
      .#environment = "staging"
      log("end")
      ""
```
Then it looks like that Vector omits the #type and #environment fields. I don't see them in the vector tap output and in vector console where I send the events.  But I do see the "start" and "end" log records in the vector's console, so the transform works. 

Is it possible to add fields like `#xxxx` in the remap transform? 

P.S. I guess it is possible via add_fields transform, but as I understood it is deprecated. 

Thanks!

**Comment by spencergilbert** [comment, marked as answer]
👋 

The [docs](https://vector.dev/docs/reference/vrl/expressions/#path) note that characters that are not "alpha-numeric characters and `_`" should be quoted.

```coffee
$ ."#foo" = "bar"
"bar"

$ .
{ "#foo": "bar" }
```

So your example should be rewritten as:
```yaml
  docker_generic_logs_transform:
    type: remap
    inputs:
      - docker_db_logs_label
    source: |-
      ""
      log("start")
      ."#type" = "docker_generic"
      ."#environment" = "staging"
      log("end")
      ""
```

**Reply by podshivalovdv** [reply]
Ohh, I see, thanks a lot! I skipped that!
